File: bindings/lua/Polycode/Rotation.lua
class "Rotation"



function Rotation:__getvar(name)
	if name == "pitch" then
		return Polycore.Rotation_get_pitch(self.__ptr)
	elseif name == "yaw" then
		return Polycore.Rotation_get_yaw(self.__ptr)
	elseif name == "roll" then
		return Polycore.Rotation_get_roll(self.__ptr)
	end
end


function Rotation:__setvar(name,value)
	if name == "pitch" then
		Polycore.Rotation_set_pitch(self.__ptr, value)
		return true
	elseif name == "yaw" then
		Polycore.Rotation_set_yaw(self.__ptr, value)
		return true
	elseif name == "roll" then
		Polycore.Rotation_set_roll(self.__ptr, value)
		return true
	end
	return false
end


function Rotation:Rotation(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycore.Rotation(unpack(arg))
	end
end

function Rotation:__delete()
	if self then Polycore.delete_Rotation(self.__ptr) end
end

File: bindings/javascript/Polycode/Rotation.js
function Rotation() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.Rotation()
	}
	Object.defineProperties(this, {
		'pitch': { enumerable: true, configurable: true, get: Rotation.prototype.__get_pitch, set: Rotation.prototype.__set_pitch},
		'yaw': { enumerable: true, configurable: true, get: Rotation.prototype.__get_yaw, set: Rotation.prototype.__set_yaw},
		'roll': { enumerable: true, configurable: true, get: Rotation.prototype.__get_roll, set: Rotation.prototype.__set_roll}
	})
}

Rotation.prototype.__get_pitch = function() {
	return Polycode.Rotation__get_pitch(this.__ptr)
}

Rotation.prototype.__set_pitch = function(val) {
	Polycode.Rotation__set_pitch(this.__ptr, val)
}

Rotation.prototype.__get_yaw = function() {
	return Polycode.Rotation__get_yaw(this.__ptr)
}

Rotation.prototype.__set_yaw = function(val) {
	Polycode.Rotation__set_yaw(this.__ptr, val)
}

Rotation.prototype.__get_roll = function() {
	return Polycode.Rotation__get_roll(this.__ptr)
}

Rotation.prototype.__set_roll = function(val) {
	Polycode.Rotation__set_roll(this.__ptr, val)
}

Duktape.fin(Rotation.prototype, function (x) {
	if (x === Rotation.prototype) {
		return;
	}
	Polycode.Rotation__delete(x.__ptr)
})
